salt/modules/virt.py

```diff
@@ -2163,6 +2167,21 @@ def _remove_indent(node):
     return diff
 
 
+def _get_disk_target(targets, disks_count, prefix):
+    """
+    Compute the disk target name for a given prefix.
+
+    :param targets: the list of already computed targets
+    :param disks: the number of disks
+    :param prefix: the prefix of the target name, i.e. "hd"
+    """
+    return [
+        "{0}{1}".format(prefix, string.ascii_lowercase[i])
+        for i in range(disks_count)
+        if "{0}{1}".format(prefix, string.ascii_lowercase[i]) not in targets
+    ][0]
```

Comment:
Looks good but logically it should be more effective if we'll not build the entire list to just get the first item. Maybe something like this?
```
for i in range(disks_count):
    ret = "{0}{1}".format(prefix, string.ascii_lowercase[i])
    if ret not in targets:
        return ret
```

Reply:
right, I'll change that then